ttnn/cpp/ttnn/operations/eltwise/binary_ng/binary_ng.hpp

```diff
@@ -57,12 +57,49 @@ struct BinaryNg {
         tt::stl::Span<const unary::UnaryOpType> post_activations = {});
 };
 
+template <BinaryOpType binary_op_type>
+struct InplaceBinaryNg {
+    static Tensor invoke(
+        uint8_t queue_id,
+        const Tensor& input_tensor_a,
+        const Tensor& input_tensor_b,
+        tt::stl::Span<const unary::UnaryOpType> lhs_activations = {},
+        tt::stl::Span<const unary::UnaryOpType> rhs_activations = {},
+        tt::stl::Span<const unary::UnaryOpType> post_activations = {});
+
+    static Tensor invoke(
+        const Tensor& input_tensor_a,
+        const Tensor& input_tensor_b,
+        tt::stl::Span<const unary::UnaryOpType> lhs_activations = {},
+        tt::stl::Span<const unary::UnaryOpType> rhs_activations = {},
+        tt::stl::Span<const unary::UnaryOpType> post_activations = {});
+
+    static Tensor invoke(
+        uint8_t queue_id,
+        const Tensor& input_tensor,
+        const float scalar,
+        tt::stl::Span<const unary::UnaryOpType> lhs_activations = {},
+        tt::stl::Span<const unary::UnaryOpType> rhs_activations = {},
+        tt::stl::Span<const unary::UnaryOpType> post_activations = {});
+
+    static Tensor invoke(
+        const Tensor& input_tensor,
+        const float scalar,
+        tt::stl::Span<const unary::UnaryOpType> lhs_activations = {},
+        tt::stl::Span<const unary::UnaryOpType> rhs_activations = {},
+        tt::stl::Span<const unary::UnaryOpType> post_activations = {});
+};
+
 }  // namespace ttnn::operations::binary_ng
 
 inline Tensor typecast_to(DataType dtype, const Tensor& input) {
     return input.get_dtype() == dtype ? input : ttnn::typecast(input, dtype);
 }
 
+inline bool check_dtype(const Tensor& input) {
+    return (input.get_dtype() == DataType::BFLOAT8_B || input.get_dtype() == DataType::BFLOAT4_B);
+}
+
```

Comment:
Neither of these functions needs to be in the `hpp` file, can we move them to the `cpp`? Also `check_dtype` is not a helpful name, maybe `needs_typecast_to_bfloat16`?